app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadProgressViewModel.kt

```diff
@@ -91,6 +96,20 @@ class UploadProgressViewModel @Inject constructor(
         }
     }
 
+    fun removeAllUploadSession(onCompletion: () -> Unit) {
+        viewModelScope.launch(ioDispatcher) {
+            uploadManager.removeAllUploadSession()
+            withContext(mainDispatcher) { onCompletion() }
+        }
+    }
+
+    fun resendLastTransfer(onCompletion: () -> Unit) {
+        viewModelScope.launch(ioDispatcher) {
```

Comment:
As just above, this is not necessary because there's no blocking I/O done here at all. We can leave the default.